src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -261,7 +258,61 @@ private CaseDTO mapCaseContainerDTO(CaseContainerDTO caseDetails) {
     return caseServiceListResponse;
   }
 
-  private Optional<CaseDTO> findLatestCase(
+  private CaseDTO getLatestCaseById(UUID caseId, Boolean getCaseEvents) throws CTPException {
+
+    List<CaseDTO> cases = new ArrayList<>();
+    CaseContainerDTO caseFromRM = null;
+
+    try {
+      caseFromRM = getCaseFromRm(caseId, getCaseEvents);
+    } catch (ResponseStatusException ex) {
+      if (ex.getStatus() == HttpStatus.NOT_FOUND) {
+        log.with("caseId", caseId).debug("Case Id Not Found by Case Service");
+      } else {
+        log.with("caseId", caseId)
+            .with("status", ex.getStatus())
+            .error("Error calling Case Service");
+        throw ex;
+      }
+    }
+
+    if (caseFromRM != null) {
+      CaseDTO caseDto = mapCaseContainerDTO(caseFromRM);
+      cases.add(caseDto);
+    }
+
+    Optional<CaseDTO> cachedCase =
+        dataRepo.readCachedCaseById(caseId).map(this::createNewCachedCaseResponse);
+
+    log.with("caseId", caseId).debug("Found {} case details in Cache for caseId", cachedCase);
+
+    TimeOrderedCases timeOrderedCases = new TimeOrderedCases();
+    timeOrderedCases.add(cases);
+    Boolean cachedCaseFound = true;
```

Comment:
This is an unconventional pattern to catch NoSuchElementException, and in any case we should avoid exceptions being thrown if there is a better way of doing it.

all of these lines from here to line 301 need to be replaced with just:
```
if (cachedCase.isPresent()) {
      timeOrderedCases.addCase(cachedCase.get());
}
```

Reply:
Thanks Rob, I didn't think of using isPresent, I kept trying to use isNull but ended up with something far more convoluted than I would have liked. I have changed it now and will try to remember to use that next time.